notebooks/beir_cqadupstack.ipynb

```diff
@@ -0,0 +1,546 @@
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "#! pip install git+https://github.com/MIR-MU/pv211-utils.git"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "I0_jZfSIwfcv",
+   "metadata": {
+    "id": "I0_jZfSIwfcv",
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "### Define the necessary classes\n",
+    "\n",
+    "These will eventually represent the Queries, Documents and Relevance Judgements from the CQADupStack collection.\n",
+    "\n",
+    "Query and Document consist only of their IDs and bodies.\n",
+    "Judgements are also just a Set of Tuples that represent pairs of relevant Document-Query combinations."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "d3f3f1a8-2a5b-4d3f-9e68-175909efe906",
+   "metadata": {
+    "id": "d3f3f1a8-2a5b-4d3f-9e68-175909efe906",
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "from pv211_utils.beir.entities import BeirDocumentBase, BeirQueryBase, BeirJudgementBase\n",
+    "from typing import Set\n",
+    "\n",
+    "class Query(BeirQueryBase):\n",
+    "    \"\"\"\n",
+    "    A processed query form the Beir collection.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    query_id : int\n",
+    "        The number\n",
+    "    body : str\n",
+    "        Text of a query\n",
+    "\n",
+    "    \"\"\"\n",
+    "    def __init__(self, query_id: int, body: str):\n",
+    "        super().__init__(query_id, body)\n",
+    "\n",
+    "class Document(BeirDocumentBase):\n",
+    "    \"\"\"\n",
+    "    A processed document form the Beir collection.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    document_id : str\n",
+    "        A unique identifier of the document.\n",
+    "    body : str\n",
+    "        The text of the document.\n",
+    "\n",
+    "    \"\"\"\n",
+    "    def __init__(self, document_id: str, body: str):\n",
+    "        super().__init__(document_id, body)\n",
+    "        \n",
+    "BeirJudgements = Set[BeirJudgementBase]\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "HQEFV7C5wtZv",
+   "metadata": {
+    "id": "HQEFV7C5wtZv",
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "### Define the datasets that are to be used.\n",
+    "\n",
+    "\n",
+    "TODO - Either this or the one bellow\n",
+    "\n",
+    "RawBeirDataset stores the basic setup \n",
+    "- Name of the dataset\n",
+    "- Subset(s) to use\n",
+    "- Alternative(s) to the subset(s) if they are not available\n",
+    "\n",
+    "\n",
+    "RawBeirDatasets then stores \n",
+    "- Common download location\n",
+    "- List of RawBeirDataset instances\n",
+    "\n",
+    "If more than one datasets are used, they will be merged and used as one. This functionality is primarly aimed at the CQADupstack datasets, but shuld work with any other combination as well."
```

Comment:
The text of the notebook should tell the user what they need to know for their second term project, no-nonsense. The discussion of loading different BEIR datasets should be moved to the docstring of `beir/eval.py` from which we can automatically generate technical documentation using a tool such as `sphinx` and/or to a separate notebook. Here, it is just confusing.

Reply:
todo